### PR TITLE
[feature] timeslot

### LINKED ISSUE #375: [feature] Timeslot mechanic
- Given 3 per hari: pagi, siang, malem
- Interact dengan sesuatu (dummy) ke progress time slot nya
- Kalo dari malem di interact lanjut jadi next day pagi
- Implemen HUD sementara untuk day time nya (hari tanggal timeslot: morning, afternoon, evening)
- Implemen interact yang dungeon (dummy) juga (Makan timeslot pagi + siang | siang + malem)

### PR DESCRIPTION
**Pull request by rafiftaris**
### Main changes
- Implemented TimeslotStateMachine as a Timeslot manager
- Implemented dummy interactable that can advance time when interacted
- Implemented dummy hud and integrate it with TimeslotStateMachine

### Tasks List
- [close #375 ]

### Attachments

https://user-images.githubusercontent.com/51603162/182403702-4ccb222d-d7e7-45c0-9bc7-d5d1978fdbe4.mp4

### CODE REVIEW

**Review by RaiRiz — APPROVED** [review]

**Review by VadimBhirawa — APPROVED** [review]
The code looks good enough to me with the state machine, good job!

**Review by RaiRiz — APPROVED** [review]

File: Assets/Astralization/Scripts/Interactables/DummyTimeslotChanger/DummyTimeslotChanger.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    #region Events
+    public static event Action<int> TimeJumpEvent;
+    #endregion
+
+    #region Variable
+    [SerializeField]
+    [Range(1, 2)]
+    [Tooltip("Timeslot jump count when interacted")]
+    private int _timeJumpCount = 1;
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/Interactables/DummyTimeslotChanger/DummyTimeslotChanger.cs:16]: Rename variable: _timeslotIncrement

File: Assets/Astralization/Scripts/Interactables/DummyTimeslotChanger/DummyTimeslotChanger.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DummyTimeslotChanger : Interactable
+{
+    #region Events
+    public static event Action<int> TimeJumpEvent;
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/Interactables/DummyTimeslotChanger/DummyTimeslotChanger.cs:9]: Rename event: TimeslotIncrementEvent

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 3]
```diff
@@ -93,5 +93,10 @@ public void ResetPlayerState()
         SendPlayerStateEvent(Utils.PlayerHelper.States.Default);
         //Debug.Log("[MANAGER] Reset player state to default");
     }
+
+    public void InvokeTimeJump(int jumpCount)
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:97]: Rename function: InvokeTimeslotIncrement

File: Assets/Astralization/Scripts/Managers/HudManager.cs [hunk 4]
```diff
@@ -89,5 +93,10 @@ private void HandleInventoryEvent(InventoryHudEventArgs args)
                 break;
         }
     }
+
+    private void UpdateTimeDisplay(DateTimeslot dateTimeslot)
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/Managers/HudManager.cs:97]: Rename function: UpdateTimeslotDisplay

File: Assets/Astralization/Scripts/States/TimeslotState/TimeslotStateMachine.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+public struct DateTimeslot
+{
+    public DateTime Date;
+    public TimeslotState Timeslot;
+}
+
+public class TimeslotStateMachine : StateMachine
+{
+    #region Events
+    public static event Action<DateTimeslot> UpdateTimeHudEvent;
```
Comment (InShineStudios) [inline Assets/Astralization/Scripts/States/TimeslotState/TimeslotStateMachine.cs:15]: Rename event: UpdateTimeslotHudEvent

**Review by InShineStudios — CHANGES_REQUESTED** [review]

**Review by InShineStudios — APPROVED** [review]

**Pull request merged**